### PR TITLE
[NFC] github: Add code owners for some components

### PR DESCRIPTION
**Pull request by AnthonyLatsis**
I’m an external contributor, so this is highly subjective and unrefined. Please share your thoughts and let me know if you want to be included or excluded from anywhere, or if I’ve missed something! Especially you, Suyash, since I haven’t included you anywhere yet.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -2,6 +2,15 @@
 # Each line is a file pattern followed by one or more owners.
 # Order is important. The last matching pattern has the most precedence.
 
+# .github
+/.github/                         @shahmishal
+/.github/ISSUE_TEMPLATE/          @AnthonyLatsis @hborla @LucianoPAlmeida @shahmishal @xedin
+/.github/PULL_REQUEST_TEMPLATE.md @AnthonyLatsis @hborla @LucianoPAlmeida @shahmishal @xedin
+
+# docs
+/docs/HowToGuides/ @AnthonyLatsis @LucianoPAlmeida
```
Comment (xedin) [inline .github/CODEOWNERS:11]: Could you please add me to this one as well?

**Review by xedin — APPROVED** [review]
Looks like a reasonable start!

File: .github/CODEOWNERS (hunk 1 above)
Comment (AnthonyLatsis) [reply .github/CODEOWNERS:11]: Sure!

Comment (xedin) [reply .github/CODEOWNERS:11]: Thank you!

**Comment by ktoso** [comment]
If we'd be able to I'd love to utilize this to be notified about all `Distributed` changes, though as they're pretty spread out, it'd have to cover all these:

```
./include/swift/AST/DistributedDecl.h
./include/swift/SILOptimizer/Utils/DistributedActor.h
./lib/AST/DistributedDecl.cpp
./lib/IRGen/GenDistributed.cpp
./lib/IRGen/GenDistributed.h
./lib/SILGen/SILGenDistributed.cpp
./lib/SILOptimizer/Utils/DistributedActor.cpp
./lib/Sema/CodeSynthesisDistributedActor.cpp
./lib/Sema/DerivedConformanceDistributedActor.cpp
./lib/Sema/TypeCheckDistributed.cpp
./lib/Sema/TypeCheckDistributed.h
```

can we glob this like `lib/Sema/.*Distributed.*`, `lib/IRGen/.*Distributed.*` etc... And the simple ones like `stdlib/public/Distributed/` `test/Distributed`. Though I can probably add myself later.

**Comment by AnthonyLatsis** [comment]
@ktoso No worries, I'll see to it. Thanks for reacting!

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -2,6 +2,15 @@
 # Each line is a file pattern followed by one or more owners.
 # Order is important. The last matching pattern has the most precedence.
```
[the remark this answers is not in the document]
Comment (AnthonyLatsis) [reply .github/CODEOWNERS:3]: The small clarification and link make sense. I think further elaborations are overkill; we don’t want this to be a pattern format reference — people can refer to the article with those kind of questions.

**Comment by AnthonyLatsis** [comment]
> just asking if such globbing would work or not

Sorry, missed this part. Yes, that will work, except in a slightly different format, like a `.gitignore`.

**Comment by theblixguy** [comment]
> Especially you, Suyash, since I haven’t included you anywhere yet.

For now I don't want to be mentioned anywhere, but perhaps in the near future I will request to be added to some files 😄

**Comment by ktoso** [comment]
Thank you, this looks very good :) The rationale on how we add those also makes sense, thanks!

**Review by ktoso — APPROVED** [review]

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -29,3 +48,84 @@ cmake/**/*Windows*    @compnerd
...
+# TODO: /include/swift/SIL/
+# TODO: /lib/SIL/
+# TODO: /lib/SILGen/
+/lib/SILGen/*Distributed* @ktoso
+/test/Distributed/SIL/    @ktoso
+# TODO: /*test/SIL/
+# TODO: /*test/SILGen/
+# FIXME: This file should go inside either 'test/Distributed/SIL' or
+# 'test/SILGen/Distributed'.
+/test/SILGen/distributed_thunk.swift @ktoso
```
Comment (xedin) [inline .github/CODEOWNERS:114]: @ktoso Could you please move this file? It seems like SILGen/Distributed/ is a good spot even if it's just one file (for now).

File: .github/CODEOWNERS [hunk 4]
```diff
@@ -29,3 +48,84 @@ cmake/**/*Windows*    @compnerd
 lib/Basic/Windows     @compnerd
 stdlib/public/Windows @compnerd
 utils/*windows*       @compnerd
+
+# AST
+/include/swift/AST/               @hborla @slavapestov @xedin
+/include/swift/AST/*Conformance*  @slavapestov
+/include/swift/AST/*Distributed*  @ktoso
+/include/swift/AST/Evaluator*     @CodaFi @slavapestov
```
Comment (xedin) [inline .github/CODEOWNERS:56]: @CodaFi Do you want to be mentioned to any of these?

**Review by xedin — APPROVED** [review]
Makes sense to me, thank you!

File: .github/CODEOWNERS (hunk 3 above)
Comment (ktoso) [reply .github/CODEOWNERS:114]: Sure, I'll move it 👍

Comment (ktoso) [reply .github/CODEOWNERS:114]: Done in: https://github.com/apple/swift/pull/62357 it'll be `test/Distributed/SIL/distributed_thunk.swift`

Comment (xedin) [reply .github/CODEOWNERS:114]: Thank you!

**Comment by AnthonyLatsis** [comment]
@slavapestov Do you have any objections concerning the areas I have assigned you to?

**Comment by AnthonyLatsis** [comment]
@hborla ping

File: .github/CODEOWNERS (hunk 4 above)
Comment (CodaFi) [reply .github/CODEOWNERS:56]: Works for me

**Review by CodaFi — APPROVED** [review]

**Review by hborla — APPROVED** [review]

**Comment by AnthonyLatsis** [comment]
@swift-ci please smoke test

**Pull request merged**